packages/cli/src/plugin.js

```diff
@@ -0,0 +1,256 @@
+import fs from 'fs'
+import path from 'path'
+
+import chalk from 'chalk'
+
+import { getConfig, getPaths } from './lib'
+import { installModule } from './lib/packages'
+
+const PLUGIN_CACHE_FILENAME = 'command-cache.json'
+
+function checkPluginListAndWarn(plugins) {
+  // Plugins must define a package and version
+  for (const plugin of plugins) {
+    if (!plugin.package) {
+      console.warn(
+        chalk.yellow(
+          `⚠️  A plugin is missing a package name, it cannot be loaded.`
+        )
+      )
+    }
+    if (!plugin.version) {
+      console.warn(
+        chalk.yellow(
+          `⚠️  Plugin "${plugin.package}" is missing a version, it cannot be loaded.`
+        )
+      )
+    }
+  }
+
+  // Plugins should only occur once in the list
+  const pluginNames = plugins.map((p) => p.package)
+  if (plugins.length !== new Set(pluginNames).size) {
+    console.warn(
+      chalk.yellow(
+        '⚠️  Duplicate plugin names found in redwood.toml, duplicates will be ignored.'
+      )
+    )
+  }
+
+  // Plugins should be published to npm under a scope which is used as the namespace
+  const namespaces = plugins.map((p) => p.package.split('/')[0])
+  namespaces.forEach((ns) => {
+    if (!ns.startsWith('@')) {
+      console.warn(
+        chalk.yellow(
+          `⚠️  Plugin "${ns}" is missing a scope/namespace, it will not be loaded.`
+        )
+      )
+    }
+  })
+}
+
+export async function loadPlugins(yargs) {
+  const { plugins, autoInstall } = getConfig().experimental.cli
+
+  const enabledPlugins = plugins.filter((p) => p.enabled ?? true)
+
+  // Print warnings about invalid plugins
+  checkPluginListAndWarn(enabledPlugins)
+
+  const redwoodPackages = new Set()
+  const thirdPartyPackages = new Set()
+  for (const plugin of enabledPlugins) {
+    // Skip invalid plugins
+    if (!plugin.package || !plugin.version) {
+      continue
+    }
+    // Skip non-scoped packages
+    if (!plugin.package.startsWith('@')) {
+      continue
+    }
+    if (plugin.package.startsWith('@redwoodjs/')) {
+      redwoodPackages.add(plugin.package)
+    } else {
+      thirdPartyPackages.add(plugin.package)
+    }
+  }
+
+  // Order alphabetically but with @redwoodjs packages first
+  const namespaces = Array.from(
+    redwoodPackages.map((p) => p.split('/')[0])
+  ).sort()
```

Comment:
The array is just one member-long at this point right? So do we need to run `sort`? And wouldn't there always be one RedwoodJS namespace? (So is the map useful?) You may have something in mind long term here though

```suggestion
  const namespaces = Array.from(
    redwoodPackages.map((p) => p.split('/')[0])
  )
```